Source/Launcher/Program.cs

```diff
@@ -37,59 +36,116 @@ namespace ORTS
 {
 	static class Program
 	{
-		/// <summary>
-		/// The main entry point for the application.
-		/// </summary>
-		[STAThread]
-		static void Main()
+        static string downloadUrl;
+        static string downloadText;
+        static List<string>missing = new List<string>();
+
+        /// <summary>
+        /// The main entry point for the application.
+        /// </summary>
+        [STAThread]
+        static void Main()
 		{
 			Application.EnableVisualStyles();
 
 			// Check for any missing components.
 			var path = Path.GetDirectoryName(Application.ExecutablePath);
-			var missing = new List<string>();
-			CheckNetFx(missing);
-			CheckXNA(missing);
-			CheckOR(missing, path);
-			if (missing.Count > 0)
+
+            CheckNetFx();
+			CheckXNA();
+            CheckDXRuntime();
+            CheckOpenAL();
```

Comment:
Likewise, not passing in the `missing` list here is strange to me. Functions with side effects like this are harder to understand.

Reply:
I don't agree, from semantics passing as variables within a static class does come at a cost, and if there would be any side effects, something else is inherently broken. 
But given the small code size of the artifact and for code clarity, I've updated the code.

Reply:
The side effect here is the method modifying the static field (it's not mentioned on these lines so why would you expect it to change or even know it exists?), which is unnecessary and confusing when reading the code. Thank you for changing it!